base/tps/src/main/java/org/dogtagpki/server/tps/rest/base/ConnectorProcessor.java

```diff
@@ -0,0 +1,465 @@
+
+            ConnectorData connectorData = createConnectorData(database.getRecord(connectorID));
+            auditModParams.put("Status", status);
+            auditTPSConnectorChange(principal, ILogger.SUCCESS, method, connectorData.getID(), auditModParams, null);
+
+            return connectorData;
+
+        } catch (PKIException e) {
+            logger.error("ConnectorProcessor: " + e.getMessage(), e);
+            auditTPSConnectorChange(principal, ILogger.FAILURE, method,
+                    connectorID, auditModParams, e.toString());
+            throw e;
+
+        } catch (Exception e) {
+            logger.error("ConnectorProcessor: " + e.getMessage(), e);
+            auditTPSConnectorChange(principal, ILogger.FAILURE, method,
+                    connectorID, auditModParams, e.toString());
+            throw new PKIException(e);
+        }
+    }
+
+    public void removeConnector(Principal principal, String connectorID) {
+        String method = "ConnectorProcessor.removeConnector";
+        Map<String, String> auditModParams = new HashMap<>();
+
+        if (connectorID == null) {
+            auditConfigTokenGeneral(principal, ILogger.FAILURE, method, null,
+                    "Connector ID is null.");
+            throw new BadRequestException("Connector ID is null.");
+        }
+        auditModParams.put("connectorID", connectorID);
+
+        logger.debug("ConnectorProcessor.removeConnector(\"{}\")", connectorID);
+
+        try {
+            ConnectorRecord connRecord = database.getRecord(connectorID);
+            String status = connRecord.getStatus();
+
+            if (!Constants.CFG_DISABLED.equals(status)) {
+                Exception e = new ForbiddenException("Unable to delete connector "
+                        + connectorID
+                        + "; connector not disabled");
+                auditTPSConnectorChange(principal, ILogger.FAILURE, method,
+                        connectorID, auditModParams, e.toString());
+                throw e;
+            }
+
+            database.removeRecord(connectorID);
+            auditTPSConnectorChange(principal, ILogger.SUCCESS, method, connectorID, null, null);
+
+        } catch (PKIException e) {
+            logger.error("ConnectorProcessor: " + e.getMessage(), e);
+            auditTPSConnectorChange(principal, ILogger.FAILURE, method,
+                    connectorID, auditModParams, e.toString());
+            throw e;
+
+        } catch (Exception e) {
+            logger.error("ConnectorProcessor: " + e.getMessage(), e);
+            auditTPSConnectorChange(principal, ILogger.FAILURE, method,
+                    connectorID, auditModParams, e.toString());
+            throw new PKIException(e);
+        }
+    }
+
+    private ConnectorData createConnectorData(ConnectorRecord connectionRecord) throws UnsupportedEncodingException {
+
+        String connectorID = connectionRecord.getID();
+        URLEncoder.encode(connectorID, "UTF-8");
```

Comment:
The original code that calls `URLEncoder.encode()` seems to be redundant, so we probably can remove this line. What do you think?